File: cpp/Motherboard.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;

class Motherboard : public Komponen
{
private:
    int jumlah_slot_RAM;
    int jumlah_slot_Penyimpanan;

public:
    Motherboard() {}

    Motherboard(int jumlah_slot_RAM, int jumlah_slot_Penyimpanan, string nama, string merk, int tahun_rilis, int harga) : Komponen(nama, merk, tahun_rilis, harga)
    {
        this->jumlah_slot_RAM = jumlah_slot_RAM;
        this->jumlah_slot_Penyimpanan = jumlah_slot_Penyimpanan;
    }

    void setJumlahSlotRAM(int jumlah_slot_RAM)
    {
        this->jumlah_slot_RAM = jumlah_slot_RAM;
    }

    void setJumlahSlotPenyimpanan(int jumlah_slot_Penyimpanan)
    {
        this->jumlah_slot_Penyimpanan = jumlah_slot_Penyimpanan;
    }

    int getJumlahSlotRAM()
    {
        return jumlah_slot_RAM;
    }

    int getJumlahSlotPenyimpanan()
    {
        return jumlah_slot_Penyimpanan;
    }

    ~Motherboard() {}
};

File: python/Motherboard.py
from Komponen import Komponen

class Motherboard(Komponen):
    def __init__(self, jumlah_slot_RAM=0, jumlah_slot_Penyimpanan=0, nama="", merk="", tahun_rilis=0, harga=0):
        super().__init__(nama, merk, tahun_rilis, harga)
        self.jumlah_slot_RAM = jumlah_slot_RAM
        self.jumlah_slot_Penyimpanan = jumlah_slot_Penyimpanan
    
    def setJumlahSlotRAM(self, jumlah_slot_RAM):
        self.jumlah_slot_RAM = jumlah_slot_RAM
    
    def setJumlahSlotPenyimpanan(self, jumlah_slot_Penyimpanan):
        self.jumlah_slot_Penyimpanan = jumlah_slot_Penyimpanan
    
    def getJumlahSlotRAM(self):
        return self.jumlah_slot_RAM
    
    def getJumlahSlotPenyimpanan(self):
        return self.jumlah_slot_Penyimpanan